src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

```diff
@@ -49,6 +62,27 @@ public class XdocsTemplateSink extends XdocSink {
     public XdocsTemplateSink(Writer writer, String encoding) {
         super(writer);
         this.encoding = encoding;
+        this.writer = new PrintWriter(writer);
```

Comment:
Fact that we put in super one instance of writer and in this class create another instance of writer.
It is error prone.
We need to create method `PrintWriter initWtiter(writer) { this.writer = new PrintWriter(writer); return this.wrieter;}`

Reply:
This is a good point. Really, we can probably just avoid all the other hacks in here and just do:
```java
    private static final class MyPrintWriter extends PrintWriter {
        private MyPrintWriter(Writer writer) {
            super(writer);
        }

        @Override
        public void println() {
            write("\n");
        }
    }
```

then:
```
super(new MyWriter(writer));
```

Reply:
In https://issues.apache.org/jira/browse/DOXIA-707 maintainer just mentioned this approach to provide writer that does \n on line end.

Reply:
And in https://github.com/checkstyle/checkstyle/issues/13770#issuecomment-1752135819